src/System.Private.CoreLib/src/System/Numerics/Hashing/HashHelpers.cs

```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Numerics.Hashing
+{
+    // Please change the corresponding file in corefx if this is changed.
+
+    internal static class HashHelpers
+    {
+        public static int Combine(int h1, int h2)
+        {
+            uint shift5 = ((uint)h1 << 5) | ((uint)h1 >> 27);
```

Comment:
I think the comment about optimizations was useful here - you can tweak it to not specifically talk about JIT, e.g. `// This should get optimized to use a single ROL instruction`.